src/YumeChan.Core/Services/Plugins/NugetPluginsFetcher.cs

```diff
@@ -0,0 +1,340 @@
+			ClientPolicyContext.GetClientPolicy(nugetSettings, NullLogger.Instance), NullLogger.Instance);
+
+		foreach ((SourcePackageDependencyInfo package, bool isPlugin) in packagesToInstall)
+		{
+			DownloadResource? downloadResource = await package.Source.GetResourceAsync<DownloadResource>(ct);
+			
+			// Download the package (or fetch it from the cache).
+			DownloadResourceResult downloadResult = await downloadResource.GetDownloadResourceResultAsync(
+				package, new(_sourceCacheContext), SettingsUtility.GetGlobalPackagesFolder(nugetSettings), NullLogger.Instance, ct);
+			
+			// Extract the package into the target directory.
+			await PackageExtractor.ExtractPackageAsync(downloadResult.PackageSource, downloadResult.PackageStream, 
+				isPlugin ? pluginPackagesPathResolver : dependenciesPathResolver,
+				packageExtractionContext, ct);
+		}
+	}
+
+	private async Task InstallPluginPackagesAsync(PackageIdentity plugin, IEnumerable<SourcePackageDependencyInfo> packages, string rootPluginsDirectory, ISettings nugetSettings, CancellationToken ct)
+	{
+		PluginDependenciesPathResolver pluginDependenciesPathResolver = new(rootPluginsDirectory, plugin.Id);
+		
+		PackageExtractionContext packageExtractionContext = new(
+			PackageSaveMode.Files,
+			XmlDocFileSaveMode.Skip,
+			ClientPolicyContext.GetClientPolicy(nugetSettings, NullLogger.Instance), 
+			NullLogger.Instance);
+
+		foreach (SourcePackageDependencyInfo package in packages)
+		{
+			DownloadResource? downloadResource = await package.Source.GetResourceAsync<DownloadResource>(ct);
+			
+			// Download the package (or fetch it from the cache).
+			DownloadResourceResult downloadResult = await downloadResource.GetDownloadResourceResultAsync(
+				package, new(_sourceCacheContext), SettingsUtility.GetGlobalPackagesFolder(nugetSettings), NullLogger.Instance, ct);
+			
+			// Extract the package into the target directory.
+			await PackageExtractor.ExtractPackageAsync(downloadResult.PackageSource, downloadResult.PackageStream, 
+				pluginDependenciesPathResolver,
+				packageExtractionContext, ct);
+		}
+	}
+	
+	public void Dispose()
+	{
+		_sourceCacheContext.Dispose();
+		GC.SuppressFinalize(this);
+	}
+	
+	private static bool DependencySuppliedByHost(DependencyContext hostDependencies, PackageDependency dep)
+	{
+		// See if a runtime library with the same ID as the package is available in the host's runtime libraries.
+		RuntimeLibrary? runtimeLib = hostDependencies.RuntimeLibraries.FirstOrDefault(r => r.Name == dep.Id);
+ 
+		// If not, does it exist as a .dll file, in the host's runtime directory?
+		if (runtimeLib is null)
+		{
+			if (File.Exists(Path.Combine(Assembly.GetExecutingAssembly().Location, dep.Id + ".dll")))
+			{
+				return true;
+			}
+		}
+		
+		return runtimeLib is not null;
+	}
+	
+	private static async Task FlattenDownloadedPackageToDirectoryStructureAsync(DirectoryInfo downloadFolder, DirectoryInfo targetFolder, CancellationToken ct = default)
```

Comment:
```suggestion
	private static Task FlattenDownloadedPackageToDirectoryStructureAsync(DirectoryInfo downloadFolder, DirectoryInfo targetFolder, CancellationToken ct = default)
```
🦤

![image](https://user-images.githubusercontent.com/2266487/162284814-4a7da5e6-d26b-4f80-a6a4-2f235ef3399b.png)

Reply:
Fixed in 7ce54622ab8ef62bd4b617e4fbd9c175342ff156.